tools/src/main/java/org/apache/kafka/tools/consumer/ConsoleConsumerOptions.java

```diff
@@ -238,7 +237,8 @@ private Set<String> checkConsumerGroup(Properties consumerPropsFromFile, Propert
     }
 
     private Properties buildConsumerProps(Properties consumerPropsFromFile, Properties extraConsumerProps, Set<String> groupIdsProvided) {
-        Properties consumerProps = new Properties(consumerPropsFromFile);
+        Properties consumerProps = new Properties();
+        consumerProps.putAll(consumerPropsFromFile);
```

Comment:
Strange that this change is not doing the exact same thing. But reading the API, constructor with `Property` says that it creates an empty list with the passed properties as defaults. Tricky API design from Java's part...

Reply:
Yep! Default values are only retrieved when using `getProperty()` and not when using `get()`. This is why the unit test was passing.